### PR TITLE
MM-45334 : Custom status not setting over the websocket

### PR DESCRIPTION
**Pull request by M-ZubairAhmed**
#### Summary
Added few missing fields of user object in gql query

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-45334

#### Related Pull Requests
Server PR https://github.com/mattermost/mattermost-server/pull/20646

#### Screenshots
N/A

#### Release Note

```release-note
NONE
```

### CODE REVIEW

File: packages/mattermost-redux/src/actions/users_queries.ts [hunk 1]
```diff
@@ -66,7 +73,7 @@ query MyData {
       scheme_user: schemeUser
       scheme_admin: schemeAdmin
     }
-}
+  }
 `;
 
 export const myDataQuery = JSON.stringify({query: myDataQueryString, operationName: 'MyData'});
```
Comment (agnivade) [inline packages/mattermost-redux/src/actions/users_queries.ts:79]: The operation name here is what needs to be changed :)

In any case, what I suggested was to rename `myData` to something more readable and intuitive. You have the word `myData` throughout this file - `myDataQueryString`, `MyDataQueryResponseType` etc. You'll have more and more gql queries, so we need names that correspond to what it serves in the UI. Maybe `gql-w-userInfo` is a better name? 0/5.

**Review by agnivade — CHANGES_REQUESTED** [review]

File: packages/mattermost-redux/src/actions/users_queries.ts (hunk 1 above)
Comment (M-ZubairAhmed) [reply packages/mattermost-redux/src/actions/users_queries.ts:79]: oh forgot about the operation name only :P

File: packages/mattermost-redux/src/actions/users_queries.ts [hunk 2]
```diff
@@ -114,17 +121,6 @@ export function transformToRecievedMeReducerPayload(user: Partial<MyDataQueryRes
...
-        // below fields arent included in the response but not inside of user rest api types
-        auth_data: '',
-        email_verified: true,
-        allow_marketing: true,
-        last_activity_at: 0,
-        bot_description: '',
-        bot_last_icon_update: 0,
-        terms_of_service_id: '',
-        terms_of_service_create_at: 0,
-        remote_id: '',
```
Comment (M-ZubairAhmed) [inline packages/mattermost-redux/src/actions/users_queries.ts:127]: Fields such as auth_dat, email_verified, allow_marketing, bot_last_icon_update are no used in the webapp, hence removing them from the query. They are available in the gql, if anyone would like to add those in the future

**Review by agnivade — APPROVED** [review]

File: packages/mattermost-redux/src/actions/users_queries.ts (hunk 2 above)
Comment (hmhealey) [reply packages/mattermost-redux/src/actions/users_queries.ts:127]: Could you remove those from the type definition for `User` on the client then? Let's make sure people don't start relying on them by mistake

**Review by hmhealey — CHANGES_REQUESTED** [review]

File: packages/mattermost-redux/src/actions/users_queries.ts (hunk 2 above)
Comment (hmhealey) [reply packages/mattermost-redux/src/actions/users_queries.ts:127]: You'll also probably need to remove them from the TestHelper method that generates a fake user object and update the snapshots that use that method as well too then

Comment (M-ZubairAhmed) [reply packages/mattermost-redux/src/actions/users_queries.ts:127]: Cross checked all unnecessary types from `User` and removed the ones not used.

**Review by hmhealey — APPROVED** [review]

**Comment by jgilliam17** [comment]
e2es failed to trigger, updating the branch and trying again

**Comment by jgilliam17** [comment]
Thanks @M-ZubairAhmed 
Tested, looks good to merge.
Verified custom status sets immediately, no refresh needed.
[E2E report](https://mattermost-cypress-report.s3.amazonaws.com/223625-a62993b-onprem-ent-webapp-pr-10796-mattermost/mm-ee-testa62993b/mochawesome.html) - no PR related failures.

**Review by jgilliam17 — APPROVED** [review]

**Pull request merged**